components/feature/session/src/main/java/mozilla/components/feature/session/SessionUseCases.kt

```diff
@@ -42,7 +44,8 @@ class SessionUseCases(
          * @param url The URL to be loaded using the selected session.
          */
         override fun invoke(url: String) {
-            this.invoke(url, sessionManager.selectedSession)
+            val loadSession = if (useNewSession) onNoSession.invoke(url) else sessionManager.selectedSession
+            this.invoke(url, loadSession)
```

Comment:
This is already covered by the `TabsUseCases` (`AddNewTabUseCase`, `AddNewPrivateTabUseCase`) and they implement `LoadUrlUseCase` so that you can use them instead of the `DefaultLoadUrlUseCase` here.